spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/messaging/TraceChannelInterceptorTests.java

```diff
@@ -194,6 +196,12 @@ public void parentSpanIncluded() {
 		then(traceId).isEqualTo(10L);
 		then(spanId).isNotEqualTo(20L);
 		then(this.accumulator.getSpans()).hasSize(1);
+		List<Log> logs = this.accumulator.getSpans().get(0).logs();
+		then(logs).hasSize(4);
+		then(logs.get(0).getEvent()).isEqualTo(Span.CLIENT_SEND);
+		then(logs.get(1).getEvent()).isEqualTo(Span.SERVER_RECV);
+		then(logs.get(2).getEvent()).isEqualTo(Span.SERVER_SEND);
+		then(logs.get(3).getEvent()).isEqualTo(Span.CLIENT_RECV);
```

Comment:
Yeah... That's cool! That's exactly what I would like to see in the test-case. 😄 
Although these names are a bit misleading. Anyway I think this now maps directly to the:
```
preSend -> beforeHandle -> afterHandled -> postSend(afterSendComletion)
```

And this should be exactly the same what we have with the `ExecutorChannel`.
Although there thanks to the thread shifting we have a slightly different model:
```
preSend -> postSend(afterSendComletion)
      beforeHandle -> afterHandled
```

Anyway I think that is exactly what we are expecting from such a behavior!

Reply:
Yeah... the test was wrong. It wasn't testing enough. Now it's testing that: 

- the message was sent and waiting for response (Client Sent)
- the message was received and started processing (Server Received)
- the message was received and finished processing (Server Sent)
- the message was sent and response was received (Client Received)

Before the change no "Server side" was checked